diff --git a/sw/nic/gpuagent/Makefile b/sw/nic/gpuagent/Makefile
index f1c30b7..3f7f82b 100644
--- a/sw/nic/gpuagent/Makefile
+++ b/sw/nic/gpuagent/Makefile
@@ -131,7 +131,7 @@ LDFLAGS_COMMON  := -pthread -rdynamic -no-canonical-prefixes -Wl,--gc-sections \
                    -Wl,-z,relro,-z,now -Wl,--build-id=md5 -Wl,--hash-style=gnu \
                    -L$(BLD_LIB_DIR) -Wl,-rpath,$(BLD_LIB_DIR) \
 				   -Wl,-rpath-link,$(BLD_LIB_DIR) -L${BLD_DIR}/lib -L${BLD_DIR}/lib64 \
-                   -l:libprotobuf.a -lgrpc++_reflection -lgrpc++ -lgrpc -lgpr -lupb -lcares \
+                   -l:libprotobuf.a -Wl,--start-group -lgrpc++_reflection -lgrpc++ -lgrpc -lgpr -lupb -lcares \
                    -laddress_sorting -l:libgrpc.a -l:libre2.a -labsl_synchronization -labsl_periodic_sampler -labsl_time \
                    -labsl_time_zone -labsl_int128 -labsl_bad_optional_access -labsl_bad_any_cast_impl -labsl_bad_variant_access\
                    -labsl_spinlock_wait -labsl_log_severity -labsl_strerror -labsl_raw_logging_internal -labsl_throw_delegate \
@@ -148,7 +148,7 @@ LDFLAGS_COMMON  := -pthread -rdynamic -no-canonical-prefixes -Wl,--gc-sections \
                    -labsl_statusor -labsl_flags_commandlineflag_internal -labsl_flags_commandlineflag \
                    -labsl_flags_private_handle_accessor -labsl_flags_marshalling -labsl_flags_program_name \
                    -labsl_flags_config -labsl_flags_internal -labsl_flags_reflection -labsl_flags -labsl_flags_usage_internal \
-                   -labsl_flags_usage -labsl_flags_parse -lpthread -lz -lm -lrt -ldl -l:libev.a -l:libzmq.a \
+                   -labsl_flags_usage -labsl_flags_parse -Wl,--end-group -lpthread -lz -lm -lrt -ldl -l:libev.a -l:libzmq.a \
                    -l:libssl.a -l:libcrypto.a
 LDFLAGS_AMD_SMI := ${LDFLAGS_COMMON} \
                    -L${TOPDIR}/nic/third-party/rocm/amd_smi_lib/x86_64/lib/ \
